scripts/des_y3_sne_ia/des_y3_sne_ia.py

```diff
@@ -0,0 +1,190 @@
+
+    VERSION = _VERSION
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="des_y3_sne_ia",
+            version=VERSION,
+            data_files=DataFilesPatternsDict.from_patterns({"train": ["./healpix=*/*.hdf5"]}), # This seems fairly inflexible. Probably a massive failure point.
+            description="Light curves from  DES Y3",
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = "des_y3_sne_ia"
+
+    @classmethod
+    def _info(self):
+        """Defines the features available in this dataset."""
+        # Starting with all features common to light curve datasets
+        features = {
+            'lightcurve': Sequence(feature={
+                'band': Value('string'),
+                'flux': Value('float32'),
+                'flux_err': Value('float32'),
+                'time': Value('float32'),
+            }),
+        }
+
+
+        # Adding all values from the catalog
+        for f in _FLOAT_FEATURES:
+            features[f] = Value("float32")
+        for f in _STR_FEATURES:
+            features[f] = Value("string")
+
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=Features(features),
+            # Homepage of the dataset for documentation
+            homepage=_HOMEPAGE,
+            # License for the dataset if available
+            license=_LICENSE,
+            # Citation for the dataset
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager):
+        """We handle string, list and dicts in datafiles"""
+        if not self.config.data_files:
+            raise ValueError(
+                f"At least one data file must be specified, but got data_files={self.config.data_files}"
+            )
+        data_files = dl_manager.download_and_extract(self.config.data_files)
+        if isinstance(data_files, (str, list, tuple)):
+            files = data_files
+            if isinstance(files, str):
+                files = [files]
+            # Use `dl_manager.iter_files` to skip hidden files in an extracted archive
+            files = [dl_manager.iter_files(file) for file in files]
+            return [
+                datasets.SplitGenerator(
+                    name=datasets.Split.TRAIN, gen_kwargs={"files": files}
+                )
+            ]
+        splits = []
+        for split_name, files in data_files.items():
+            if isinstance(files, str):
+                files = [files]
+            # Use `dl_manager.iter_files` to skip hidden files in an extracted archive
+            files = [dl_manager.iter_files(file) for file in files]
+            splits.append(
+                datasets.SplitGenerator(name=split_name, gen_kwargs={"files": files})
+            )
+        return splits
+
+    def _generate_examples(self, files):
```

Comment:
i've seen many `generate_examples` with an optional `object_ids` argument, e.g. https://github.com/AstroPile/AstroPile_prototype/pull/46/files#diff-8752406b3065c128d74fe897f90a753297385f758618519efdecd8c011db8be7R227

Reply:
This has now been fixed for all five surveys matching @tom-hehir solution in https://github.com/AstroPile/AstroPile_prototype/blob/3d6da624b945b1c993d8447042a23e4453ccaac0/scripts/yse/yse.py